core/src/main/java/com/linecorp/armeria/client/endpoint/WeightRampingUpStrategy.java

```diff
@@ -140,7 +140,7 @@ public EndpointSelector newSelector(EndpointGroup endpointGroup) {
             endpointGroup.addListener(newEndpoints -> {
                 // Use the executor so the order of endpoints change is guaranteed.
                 executor.execute(() -> updateEndpoints(newEndpoints));
-            });
+            }, true);
```

Comment:
How about just using listner?
```java
        RampingUpEndpointWeightSelector(EndpointGroup endpointGroup, EventExecutor executor) {
            super(endpointGroup);
            this.executor = executor;
            final AtomicBoolean initialized = new AtomicBoolean();
            endpointGroup.addListener(newEndpoints -> {
                final List<Endpoint> dedupEndpoints =
                        new ArrayList<>(deduplicateEndpoints(newEndpoints).values());
                if (!initialized.getAndSet(true)) {
                    endpointSelector = new WeightedRandomDistributionEndpointSelector(dedupEndpoints);
                } else {
                    // Use the executor so the order of endpoints change is guaranteed.
                    executor.execute(() -> updateEndpoints(newEndpoints));
                }
            }, true);
            if (endpointGroup instanceof ListenableAsyncCloseable) {
                ((ListenableAsyncCloseable) endpointGroup).whenClosed().thenRunAsync(this::close, executor);
            }
        }
```

Reply:
Sound good. Let me use `compareAndSet(false, true)` which will write a volatile value only once at the first time.